src/meta/src/stream/source_manager/split_assignment.rs

```diff
@@ -202,18 +206,18 @@ impl SourceManager {
         //
         // Note: we can choose any backfill actor to align here.
         // We use `HashMap` to dedup.
-        let aligned_actors: HashMap<ActorId, ActorId> = merge_updates
-            .values()
-            .flatten()
-            .map(|merge_update| {
-                assert_eq!(merge_update.added_upstream_actor_id.len(), 1);
-                // Note: removed_upstream_actor_id is not set for replace job, so we can't use it.
-                assert_eq!(merge_update.removed_upstream_actor_id.len(), 0);
-                (
-                    merge_update.added_upstream_actor_id[0],
-                    merge_update.actor_id,
-                )
+        let aligned_actors: HashMap<ActorId, ActorId> = new_no_shuffle
+            .iter()
+            .filter_map(|(upstream_fragment_id, new_no_shuffle)| {
+                if *upstream_fragment_id == fragment_id {
+                    Some(new_no_shuffle.values())
+                } else {
+                    None
+                }
             })
+            .flatten()
+            .flatten()
+            .map(|(upstream_actor_id, actor_id)| (*upstream_actor_id, *actor_id))
```

Comment:
worth adding some comments. It's hard to follow here
```suggestion
        // new_no_shuffle: 
        //     upstream fragment_id -> 
        //     downstream fragment_id -> 
        //     upstream actor_id -> 
        //     downstream actor_id
        let aligned_actors: HashMap<ActorId, ActorId> = new_no_shuffle
            .iter()
            .filter_map(|(upstream_fragment_id, new_no_shuffle)| {
                if *upstream_fragment_id == fragment_id {
                    Some(new_no_shuffle.values()) // Iter<upstream actor_id -> downstream actor_id>
                } else {
                    None
                }
            }) // Iter<Iter<upstream actor_id -> downstream actor_id>>
            .flatten()
            .flatten() // upstream actor_id -> downstream actor_id
            .map(|(upstream_actor_id, actor_id)| (*upstream_actor_id, *actor_id))
```

Reply:
Wait, why do we need `filter_map`? Is it just `get`..?